### ISSUE TITLE
Can't display results

### ISSUE THREAD

**Issue by neyz** [briangann/grafana-sensu-datasource#5]
Hello and thanks for your work !

I cloned the repo and can get some data to show when i select "event" source type and as well as JSON but can't seem to get any data to display for the other source. Content of the table shows "no data to display"

The "dimensions" part does get populated so i think the data is fine.

I tried playing with filters and dimensions with no luck.

Maybe i'm missing something ?

Thanks !

![capture d ecran 2017-07-07 23 46 15](https://user-images.githubusercontent.com/230362/27978269-b911ae26-636e-11e7-9f77-239e8bb13e30.png)

![capture d ecran 2017-07-07 23 47 13](https://user-images.githubusercontent.com/230362/27978273-c634b1a2-636e-11e7-8c22-5d535ddcbb81.png)

**Comment by briangann** [comment]
Not all of the source types are implemented, it's a work in progress :)

I'll get the clients source type implemented today, I need it too.

Any help is appreciated.

**Comment by neyz** [comment]
Right now I can't get any any source type to work except the JSON formatted ones that I posted. Shouldn't I at least be able to display alerts in some kind of table ? I thought I saw something like that in the readme.

Sensu version is 0.29 if that changes anything in the API results.

I don't know how to code but let me know if I can do anything.

**Comment by briangann** [comment]
I just uploaded the client implementation, give that a try.

The working sources are:

Aggregates
Aggregates JSON
    Both of the aggregate sources have 7 modes that appear when you add a dimension, and select and add an aggregate name.

Clients (this is JSON also), and you need to pick your columns

Events as JSON
Results as JSON/Table

**Comment by briangann** [comment]
You can display alerts in a table:
1) Add a table panel
2) Add new metric with sensu as the datasource
3) Select Events as JSON as the source type
4) In options choose the Table Transform JSON Data
5) Pick the columns

Here's an example:

![image](https://user-images.githubusercontent.com/7364245/27994453-6b055fea-6484-11e7-8854-9ab4a21b1aa7.png)

![image](https://user-images.githubusercontent.com/7364245/27994445-545a5f5c-6484-11e7-8955-b5698dfed6b5.png)

![image](https://user-images.githubusercontent.com/7364245/27994449-60d55cdc-6484-11e7-9396-1c80c5061a3e.png)

**Comment by neyz** [comment]
Ok great thanks for the feedback it's working for "events" and "results" !

i feel like a fool i had no idea i had to pick the columns, my bad.

Regarding the "client implementation" : i just updated with git pull and restarted grafana.

if i select "client" as source and then go to "options" - "columns" : the list is empty
same for "aggregates"

**Comment by briangann** [comment]
here's a sample for clients:

![image](https://user-images.githubusercontent.com/7364245/27994996-0a9f9f40-648d-11e7-8273-f979c68fd443.png)

![image](https://user-images.githubusercontent.com/7364245/27995003-191bab4a-648d-11e7-9332-6ce2ccabd50a.png)

**Comment by briangann** [comment]
here's a sample for aggregates (this one will need a new panel type i think)

![image](https://user-images.githubusercontent.com/7364245/27995006-3eca9ca2-648d-11e7-9f66-88d29bc8ff74.png)

![image](https://user-images.githubusercontent.com/7364245/27995014-537db620-648d-11e7-96d2-7dcf919815b0.png)

outputs:
![image](https://user-images.githubusercontent.com/7364245/27995036-5e67ba7c-648d-11e7-81df-e35213b846ad.png)

**Comment by briangann** [comment]
here is a sample for Aggregates JSON

![image](https://user-images.githubusercontent.com/7364245/27995070-d524da1e-648d-11e7-8819-847f0cd6ae7d.png)

![image](https://user-images.githubusercontent.com/7364245/27995076-dfbd6d06-648d-11e7-8d77-8a83e8c205e5.png)

**Comment by briangann** [comment]
The aggregates source type really does need a new panel type, StatusPanel might work since it handles multiple series, but it has some issues too.

I'm thinking this datasource needs to be wrapped into a Grafana App, and include sample dashboards plus custom panels.  

My goal is to not have to use Uchiwa at all, but in the short term display useful data in Grafana.  Having performance metrics (influxdb/atlas/etc), event data (sensu), and logs (elk/splunk) all in one display has been great.

**Comment by Flyweightt** [comment]
Hi Briangann,

Your datasource looks great. I am using grafana, influx, elk but now looking for events and availability checks. Sensu look promising and your data source makes it possible to intergrate it into grafana.
Besides the tables can your datasource also be used with singlestats?

Cheers

Tim

**Comment by briangann** [comment]
Yes, you can display singlestat type data, as in #Criticals, #Warnings, #Clients, and other data that comes back from the Sensu API. 

The events are best displayed in a table, I'm adding in additional metrics that will give you counts of events to help visualize the data better.

It's a work in progress, any help/input is appreciated :)
